packages/babel-plugin-class-features/src/fields.js

```diff
@@ -20,20 +24,25 @@ export function buildPrivateNamesMap(props) {
 export function buildPrivateNamesNodes(privateNamesMap, loose, state) {
   const initNodes = [];
 
-  for (const [name, { id, static: isStatic }] of privateNamesMap) {
-    // In loose mode, both static and instance fields hare transpiled using a
+  for (const [
+    name,
+    { id, static: isStatic, method: isMethod },
+  ] of privateNamesMap) {
+    // In loose mode, both static and instance fields are transpiled using a
     // secret non-enumerable property. Hence, we also need to generate that
-    // key (using the classPrivateFieldLooseKey helper) in loose mode.
+    // key (using the classPrivateFieldLooseKey helper).
     // In spec mode, only instance fields need a "private name" initializer
-    // (the WeakMap), becase static fields are directly assigned to a variable
-    // in the buildPrivateStaticFieldInitSpec function.
+    // because static fields are directly assigned to a variable in the
+    // buildPrivateStaticFieldInitSpec function.
 
     if (loose) {
       initNodes.push(
         template.statement.ast`
           var ${id} = ${state.addHelper("classPrivateFieldLooseKey")}("${name}")
         `,
       );
+    } else if (isMethod) {
```

Comment:
Can we add an assertion (it could even be a comment) that `isStatic` is false? It would make it easier to read this code.